src/Mod/Sketcher/App/Sketch.cpp

```diff
@@ -1677,6 +1677,14 @@ int Sketch::addConstraint(const Constraint *constraint)
         rtn = addPointCoincidentConstraint(constraint->First,constraint->FirstPos,constraint->Second,constraint->SecondPos);
         break;
     case PointOnObject:
+        if (Geoms[checkGeoId(constraint->Second)].type == BSpline) {
+            c.value = new double(constraint->getValue());

```

Comment:
This is acknowledged as a possible source of issues. 

I have decided that this will not be a reason not to merge this feature at this moment. Full rationale:
https://forum.freecadweb.org/viewtopic.php?p=649818#p649818